### PR TITLE
feat(Image): improve handling of HTML props

### PR DESCRIPTION
**Pull request by layershifter**
Related to #2079.

This PR improves handling of HTML props using our `partitionHTMLInputProps` function and adds examples.

### CODE REVIEW

File: src/elements/Image/Image.js [hunk 1]
```diff
@@ -73,6 +71,7 @@ function Image(props) {
     className,
   )
   const rest = getUnhandledProps(Image, props)
+  const [imgTagProps, rootProps] = partitionHTMLInputProps(rest, { htmlProps: imageProps })
```
Comment (levithomason) [inline src/elements/Image/Image.js]: This seems strange to overload an input prop parser for an image.  Should this become a generic parser?

Comment (layershifter) [reply src/elements/Image/Image.js]: Agree, I will update PR and we will go with `partitionHTMLProps`.

**Comment by layershifter** [comment]
`partitionHTMLInputProps` function was renamed to `partitionHTMLProps` . I will merge this as there was no another remarks.

**Comment by levithomason** [comment]
Released in `semantic-ui-react@0.77.1`

**Pull request merged**